sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -141,6 +140,43 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
         throw error;
       }
     }
+
+    // `createInitializedSessionReceiver` will set this value by calling init()
+    // so we just temporarily set it to "" so we can get away with it never being
+    // `undefined`.
+    this.sessionId = "";
+  }
+
+  /**
+   * Initializes the link. This method should only be called
+   * once in the lifetime of a SessionReceiver.
+   */
+  private init(): Promise<void> {
+    if (this._messageSession != null) {
+      throw new Error(
+        "Internal error: open() should not be called after the SessionReceiver has been created"
+      );
+    }
+    return this._createMessageSessionIfDoesntExist();
```

Comment:
Can we not skip the check on the message session and the following error altogether if we move `return this._createMessageSessionIfDoesntExist()` into `createInitializedSessionReceiver()`

Reply:
Ah! its because `createInitializedSessionReceiver` is static and `_createMessageSessionIfDoesntExist` is not. Then my follow up would be, do we need the static method at all? Why not create the SessionReceiverImpl as before and then call init() ?

Reply:
I can eliminate the check. I did this just so I didn't have to make `_createMessageSessionIfDoesntExist` public. 

Now that I've locked in the constructor+init() pairing I think I'll just bit the bullet and do that now.

Reply:
Done.